src/renderer/base/renderer.cpp

```diff
@@ -1241,7 +1243,7 @@ void Renderer::ResetErrorStateAndResume()
     EnablePainting();
 }
 
-void Renderer::UpdateLastHoveredInterval(const PointTree::interval newInterval)
+void Renderer::UpdateLastHoveredInterval(const std::optional<PointTree::interval> newInterval)

```

Comment:
prefer taking arguments by const reference; it reduces the number of copies the compiler needs to emit.

`const std::optional<T>& newInterval`